src/connectors/autocomplete/connectAutocomplete.ts

```diff
@@ -102,24 +102,36 @@ search.addWidgets([
     return {
       $$type: 'ais.autocomplete',
 
-      init({ instantSearchInstance, helper }) {
+      init(initOptions) {
+        const { helper, renderState, instantSearchInstance } = initOptions;
         connectorState.refine = (query: string) => {
           helper.setQuery(query).search();
         };
 
         renderFn(
           {
-            widgetParams,
-            currentRefinement: helper.state.query || '',
-            indices: [],
-            refine: connectorState.refine,
+            ...this.getWidgetRenderState!(renderState, initOptions)
+              .autocomplete!,
```

Comment:
does this gWRS not always have `autocomplete` in it?

Reply:
It doesn't have `autocomplete` until you mount an `autocomplete` widget (just like how UI state works).

Reply:
Maybe we should try to solve this later, but I'm not suer how you'd write your own widget in TS if it's not added to RenderState.

A possibility is that addWidgets has a generic, which gets filled by the widget itself, and returns a new InstantSearch type, which does have the new widget. I'm not sure, since that could be too dynamic to work in ts though

Reply:
Yep, we already had this discussion with UI state and TypeScript. Let's think about it later!